src/context.ts

```diff
@@ -1,50 +1,35 @@
-import type { Layer } from "@deck.gl/core";
+import type { UseFileUploadReturn } from "@chakra-ui/react";
+import type { Table } from "apache-arrow";
 import { createContext, type Dispatch } from "react";
-import type { StacCollection } from "stac-ts";
+import type { StacCollection, StacItem } from "stac-ts";
+import type { StacGeoparquetMetadata } from "./components/stac/stac-geoparquet";
+import type { StacValue } from "./components/stac/types";
 
-export interface AppState {
-  layer: Layer | null;
-  pickedLayer: Layer | null;
-  collections: StacCollection[];
-  selectedCollectionIds: Set<string>;
+export const StacMapContext = createContext<StacMapContextType | null>(null);
+
+interface StacMapContextType {
+  href: string | undefined;
+  setHref: (href: string | undefined) => void;
+  fileUpload: UseFileUploadReturn;
+
+  value: StacValue | undefined;
+  valueLoading: boolean;
+
+  collections: StacCollection[] | undefined;
+  collectionsLoading: boolean;
+  selectedCollections: Set<string>;
+  selectedCollectionsDispatch: Dispatch<SelectedCollectionsAction>;
+
+  stacGeoparquetTable: Table | undefined;
+  stacGeoparquetTableLoading: boolean;
+  stacGeoparquetMetadata: StacGeoparquetMetadata | undefined;
+  stacGeoparquetMetadataLoading: boolean;
+  setStacGeoparquetItemId: (id: string) => void;
+  stacGeoparquetItem: StacItem | undefined;
+  stacGeoparquetItemLoading: boolean;
```

Comment:
This feels nice, being able to describe all of the app's state in one place. Thanks for the pattern @indraneel 🙇🏼